lua/lualine/components/diff.lua

```diff
@@ -0,0 +1,183 @@
+      else
+        local min = math.min(line_stats.mod_count, line_stats.new_count)
+        modified = modified + min
+        added  = added + line_stats.new_count - min
+        removed = removed + line_stats.mod_count - min
+      end
+    end
+  end
+  git_diff = { added,  modified, removed }
+end
+
+-- variable to store git_diff getter async function
+local get_git_diff = nil
+-- flag to see if async job exited before updating git_diff
+local updated = false
+
+-- Updates the async function for current file
+local function update_git_diff_getter()
+  -- stop older function properly before overwritting it
+  if get_git_diff then get_git_diff:stop() end
+  -- Donn't show git diff when current buffer doesn't have a filename
+  if #vim.fn.expand('%') == 0 then get_git_diff = nil; git_diff=nil; return end
+  get_git_diff = async:new({
+    cmd = string.format([[git -C %s --no-pager diff --no-color --no-ext-diff -U0 -- %s]]
+    ,vim.fn.expand('%:h'), vim.fn.expand('%:t')),
+    on_stdout = function(_, data)
+      if data then
+        process_diff(data)
+        updated = true
+      end
+    end,
+    on_stderr = function (_, data)
+      if data then
+        git_diff = nil
+        updated = true
+      end
+    end,
+    on_exit = function()
+      if not updated then
+        -- updated not set means git exited without emmiting anything on stdout
+        -- or stderr means file is unchanged
+        git_diff = {0, 0, 0}
+        updated = true
+      end
+    end
+  })
+end
+
+-- Update git_diff veriable
+local function update_git_diff()
+  vim.schedule_wrap(function()
+    if get_git_diff then
+      updated = false
+      get_git_diff:start()
+    end
+  end)()
+end
+
+vim.api.nvim_exec([[
+  autocmd lualine BufEnter     * lua require'lualine.components.signify'.update_git_diff_getter()
+  autocmd lualine BufEnter     * lua require'lualine.components.signify'.update_git_diff()
+  autocmd lualine BufWritePost * lua require'lualine.components.signify'.update_git_diff()
+  ]], false)
+
+local default_color_added    = "#f0e130"
+local default_color_removed  = "#90ee90"
+local default_color_modified = "#ff0038"
+
+local function signify(options)
+  if options.colored == nil then options.colored = true end
+  -- apply colors
+  if not options.color_added then
+    options.color_added = utils.extract_highlight_colors('DiffAdd', 'guifg') or default_color_added
+  end
+  if not options.color_modified then
+    options.color_modified = utils.extract_highlight_colors('DiffChange', 'guifg') or default_color_modified
+  end
+  if not options.color_removed then
+    options.color_removed = utils.extract_highlight_colors('DiffDelete', 'guifg') or default_color_removed
+  end
+
+  local highlights = {}
+
+  -- create highlights and save highlight_name in highlights table
+  local function create_highlights()
+    highlights = {
+      highlight.create_component_highlight_group({fg = options.color_added}, 'signify_added', options),
```

Comment:
`signify_added` is no longer relevant? Maybe `lualine_diff_added` or something along those lines